spec/faraday/adapter/patron_spec.rb

```diff
@@ -15,4 +15,21 @@
 
     expect { conn.get('/') }.to raise_error(RuntimeError, 'Configuration block called')
   end
+
+  it 'reuses the patron session for keep-alive' do
+    last_session = nil
+    conn = Faraday.new do |f|
+      f.adapter :patron do |session|
+        if last_session
+          expect(session).to eq last_session
```

Comment:
Given `@config_block&.call(session)` will be called only once due to the cache, I wouldn't expect this `expect` to ever be executed.

Instead, you could use a variable to check that this block is only called once:

```ruby
    config_calls = 0
    conn = Faraday.new do |f|
      f.adapter :patron do |session|
        config_calls += 1
      end
    end
```

and then after the two `conn.get('http://example.com/')` you can

```ruby
    expect(config_calls).to eq(1)
```